### ISSUE TITLE
Exclude Wi-Fis by SSID

### ISSUE THREAD

**Issue by mjaakko** [mjaakko/NeoStumbler#296]
Currently all Wi-Fi access points are included in the reports (except those where SSID is empty or ends with "_nomap"). To save storage space and to avoid transmitting useless data over the network, those APs which get discarded by the server anyway could be excluded from the reports

**Implementation:**

Allow importing exclusion rules from a file. If no rules have been selected, all access points are collected as it is now

See https://codeberg.org/beacondb/common-network-index for the rules used by BeaconDB

**Comment by Mikaela** [comment]
I hope importing file means that a file could just be linked and it could have some local cache/fallover support if the file link returned an error.

**Comment by boredsquirrel** [comment]
Good point. I would exclude anything containing `nomap` and `optout` as people might make mistakes in their naming, like `SSID-nomap`

**Comment by mjaakko** [comment]
> It would be nice to have a list in the repo that users can easily contribute to, that is then used inside the app, somehow.

I think it's best if we use the common-network-index repo from BeaconDB and provide a way to import it to NeoStumbler. Importing it from a file would be easiest to implement, but it would be better for usability to be able to provide an URL where the updates are automatically fetched

**Comment by lucasmz-dev** [comment]
Impacts privacy as well

**Comment by boredsquirrel** [comment]
there currently is the question how to deal with SSIDs that are used for moving AND stationary wifis.

this sounds like it would require way more advanced things, like having a secondary DB of all public transport stations, and collecting those networks only when near them.

**Comment by mjaakko** [comment]
> there currently is the question how to deal with SSIDs that are used for moving AND stationary wifis.

Those can be dealt with on the server-side by looking at whether the access point is moving or not. Excluding APs based on the SSID would only be pre-filtering for those APs where we can be sure that it's moving

**Comment by matt755** [comment]
I would like to add that it would be great if a local SSID filtering could be implemented to exclude WiFi hotspots like the one of my car.
Of course, the same applies to bluetooth beacons, but as there is no "clear text information" like with SSID, I don't think it's feasible.

**Comment by lucasmz-dev** [comment]
The list by beaconDB IIRC already has some exclusions for common car names which is nice

**Comment by matt755** [comment]
Most people are too lazy to change the default settings for car hotspots, I agree. For those the server based exclusion list might work.
But in the specific case of car SSIDs an on-device filtering might also add to better privacy, as anyone who has access to the raw data and knows the SSID of the car in theory can build up a movement profile and relate it to a person.
So I still would like to see a privacy option for an on-device filtering of SSIDs. 

For a start I would wish to see the following two options implemented:
- A switch for local file import / update of an exclusion list with own entries
- A switch to import predefined URLs (beaconDB) and update them e.g. on a monthly base as suggested by @mjakko
Switch off would delete the entries.

I know, a complete implementation of the requirement would be "show me all entries, that are currently excluded" on a nice screen and also a screen based editing of entries (add, update and delete individual entries). But this, in my opinion, is not prio 1.

**Comment by joelkoen** [comment]
I've checked all SSIDs in common-network-index / optout.txt using beaconDB's data, and have pushed it here if NeoStumbler wants to fetch them to ignore client side: https://cdn.beacondb.net/config/ssid-blacklist.txt

If an SSID includes a phrase listed in that file (case sensitive), clients are encouraged to ignore that SSID by default. I was thinking clients should update this file once a day? 

I think that it should be possible to disable this list, so any APs that are incorrectly blacklisted can still be collected. It could be useful to be able 'prove' to the server that an AP is stable. Maybe in the future (once beacondb releases dumps) clients could check if an AP is included in public dumps, meaning its stable and can therefore still collect data about it with no privacy risk. But that's future stuff lol

**Comment by mjaakko** [comment]
@joelkoen Thanks! I'll try to prioritize this for the next major release

You might want to document that endpoint somewhere (same thing with the coverage layer TileJSON URL)?
